doc/building_blocks/neurovault.rst

```diff
@@ -0,0 +1,197 @@
+.. _neurovault:
+
+===========================================================
+Downloading statistical maps from the Neurovault repository
+===========================================================
+
+Neurovault is a public repository of unthresholded statistical maps,
+parcellations, and atlases of the human brain. You can read about it
+and browse the images it contains at http://www.neurovault.org. You
+can download maps from Neurovault with Nilearn.
+
+Neurovault was introduced in [1]_.
+
+Neurovault contains collections of images. We can get information
+about each collection - such as who uploaded it, a link to a paper, a
+description - and about each image - the modality, number of subjects,
+some tags, and more. The nilearn downloaders will fetch this metadata
+and the images themselves.
+
+Nilearn provides two functions to downlad statistical maps from
+Neurovault.
+
+Specific images or collections
+------------------------------
+
+In the simplest case, you already know the "id" of the collections or
+images you want. Maybe you liked a paper and went to
+http://www.neurovault.org looking for the data. Once on the relevant
+collection's webpage, you can click 'Details' to see its id
+(and more). You can then download it using
+``nilearn.datasets.fetch_neurovault_ids`` :
+
+    >>> from nilearn.datasets import fetch_neurovault_ids
+    >>> brainpedia = fetch_neurovault_ids(collection_ids=[1952]) # doctest: +SKIP
+
+Or if you want some images in particular, rather than whole
+collections :
+
+    >>> brainpedia_subset = fetch_neurovault_ids(image_ids=[32015, 32016]) # doctest: +SKIP
+
+Selection filters
+-----------------
+
+You may not know which collections or images you want. For example,
+you may be conducting a meta-analysis and want to grab all the images
+that are related to "language". Using
+``nilearn.datasets.fetch_neurovault``, you can fetch all the images and
+collections that match your criteria - you don't need to know their
+ids.
+
+The filters are applied to images' and collections' metadata.
+
+You can express filters as functions. The parameter
+``collection_filter`` should be a callable, which will be called once
+for each collection. The sole argument will be a dictionary containing
+the metadata for the collection. ``image_filter`` does the same job for
+images. The default values for these parameters don't filter out
+anything.
```

Comment:
May be we should add a note here, default values in ```image_terms``` and ```collection_terms``` need also be taken into account while using filters as functions. Because, what specify here might be contradicting with default terms in those arguments.